chain/network/Cargo.toml

```diff
@@ -4,7 +4,7 @@ version = "0.0.0"
 authors = ["Near Inc <hello@nearprotocol.com>"]
 publish = false
 # Please update rust-toolchain.toml as well when changing version here:
-rust-version = "1.56.0"
+rust-version = "1.57.0"
```

Comment:
? should this be a part of this PR? 

@matklad - can we update the rust version only for a single cargo? (as others still have 1.56, right?)

Reply:
We can, but it doesn't make sense. 

It also doesn't make sense that we have `1.57` in `rust-toolchain.toml` and `1.56` in all of the cargo-tomls. I guess we should have updated cargo.tomls as a part of https://github.com/near/nearcore/pull/5659. 

So:

* this change should be reverted from this PR
* we should send a separate PR to upgrade rust-version everywhere
  * it probably makes sense to fold this with https://github.com/near/nearcore/issues/6120
* we should patch https://github.com/near/nearcore/blob/6d86ccd4fee74c1567dc2d83d384b1952826e796/tools/themis/src/rules.rs#L28-L48 to also check that `rust-version` is the right one.

Reply:
filed https://github.com/near/nearcore/issues/6124

Reply:
To say this more explicitly: 

It's ok to actually use 1.57 features in this PR -- 1.57 is the version of Rust we use. It's a per-existing bug that Cargo.toml specifies older-than-actually-required rust version.